server/controllers/finance/journal/index.js

```diff
@@ -36,6 +40,57 @@ exports.journalEntryList = journalEntryList;
 
 exports.editTransaction = editTransaction;
 exports.count = count;
+exports.refenreceTransaction=refenreceTransaction;
+
+
+
+
+
+//loading identifiers keyies, used for defining the table name
+const identifiersIndex = {};
+function indexIdentifiers() {
+  _.forEach(identifiers, (entity) => {
+    identifiersIndex[entity.key] = entity;
+  });
+}
+
+// API GET /journal/refenrece_transaction/:codeRef/:language'
+
+//This function render a report in the browser
+//It takes a transaction reference code as paramter and the language
+//The reference code is a combination of table_key.project_abbr.reference
+//The table name is variable, it can be :invoice, cash or voucher
+
+function refenreceTransaction(req,res,next){
+   
+  //loading identifiers keyies, used for defining the table name
+  indexIdentifiers();
+
+  var codeRef =  req.params.codeRef.split(".");
+  const language= req.params.language;
+
+  const code = codeRef[0];
+  const projectName=codeRef[1];
+  const refence=codeRef[2];
+  const documentDefinition = identifiersIndex[code];
+   
+  const query = `
+    SELECT BUID(uuid) as uuid FROM ${documentDefinition.table}  tb, project p 
+    WHERE tb.project_id=p.id AND p.abbr=? AND tb.reference=?
+  `;
+  
+  // search for full UUID
+  db.one(query,[projectName,refence])
+    .then(result =>
+      {
+        let uuid=result.uuid;
+        let url= documentDefinition.documentPath + uuid + "?lang="+language+"&posReceipt=0&renderer=pdf";
```

Comment:
This string concatenation could be done quite clearly with the new String syntax

For example: 
```js
const url = `${documentDefinition.documentPath}${uuid}?lang=${language}&renderer=pdf`
```